core/src/main/java/com/linecorp/armeria/server/ServiceConfig.java

```diff
@@ -60,17 +61,22 @@
     private final Set<TransientServiceOption> transientServiceOptions;
     private final boolean handlesCorsPreflight;
 
+    @Nullable
+    private final ScheduledExecutorService blockingTaskExecutor;
+
     /**
      * Creates a new instance.
      */
     ServiceConfig(Route route, HttpService service, @Nullable String defaultLogName,
                   @Nullable String defaultServiceName, ServiceNaming defaultServiceNaming,
                   long requestTimeoutMillis, long maxRequestLength,
                   boolean verboseResponses, AccessLogWriter accessLogWriter,
-                  boolean shutdownAccessLogWriterOnStop) {
+                  boolean shutdownAccessLogWriterOnStop,
+                  @Nullable ScheduledExecutorService blockingTaskExecutor) {
```

Comment:
We can always determine the `blockingTaskExecutor` before creating a `ServiceConfig()`. How about making this parameter non-null, either by passing per-route executor or server-level executor?